spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/AppDeploymentCreateServiceInstanceWorkflow.java

```diff
@@ -108,14 +121,24 @@ public AppDeploymentCreateServiceInstanceWorkflow(BrokeredServices brokeredServi
 				credentialProviderService.addCredentials(backingApps,
 					request.getServiceInstanceId()))
 			.flatMapMany(backingApps -> deploymentService.deploy(backingApps, request.getServiceInstanceId()))
-			.doOnRequest(l -> log.debug("Deploying backing applications for {}/{}",
-				request.getServiceDefinition().getName(), request.getPlan().getName()))
-			.doOnComplete(() -> log.debug("Finished deploying backing applications for {}/{}",
-				request.getServiceDefinition().getName(), request.getPlan().getName()))
-			.doOnError(
-				exception -> log.error(String.format("Error deploying backing applications for %s/%s with error '%s'",
-					request.getServiceDefinition().getName(), request.getPlan().getName(), exception.getMessage()),
-					exception));
+			.doOnRequest(l -> {
+				LOG.info("Deploying backing applications. serviceDefinitionName={}, planName={}",
+					request.getServiceDefinition().getName(), request.getPlan().getName());
+				LOG.debug(REQUEST_LOG_TEMPLATE, request);
+			})
+			.doOnComplete(() -> {
+				LOG.info("Finish deploying backing applications. serviceDefinitionName={}, planName={}",
+					request.getServiceDefinition().getName(), request.getPlan().getName());
+				LOG.debug(REQUEST_LOG_TEMPLATE, request);
+			})
+			.doOnError(e -> {
+				if (LOG.isErrorEnabled()) {
```

Comment:
There are a few checks like this in the PR and I'm wondering if they're worth it. For debug, info levels, sure... but does anyone ever disable error level logging?

Reply:
pmd complains about not having those in some cases. We could make a choice to disable that check or live with it? Apparently some of the logging APIs have those checks built in, which is why they aren't everywhere.

Reply:
Oh, interesting - I had no idea. I don't feel strongly on this at all, it just struck me as odd that we were doing it in particular places but not others. Happy to leave as it is.

Reply:
I removed a couple of these checks, but others pmd still wanted. 🤷‍♂️